transaction-service/src/main/java/pl/adamboguszewski/transaction/service/application/Transaction.java

```diff
@@ -14,6 +14,7 @@
 public class Transaction {
 
     @Id
+    @GeneratedValue
```

Comment:
There are 4 GenerationType strategies, read about them and use best for our case.